contrib/drf_introspection/serializers.py

```diff
@@ -76,42 +112,49 @@ class DynamicFieldsSerializerMixin(object):
     query_params = ('fields', 'exclude_fields')
 
     def __init__(self, *args, **kwargs):
-        # Accept kwargs in __init__, like:
+        field_args = {}
+
         # Don't pass the 'fields' arg up to the superclass
-        fields = _pop_field_list(kwargs, 'fields')
-        exclude_fields = _pop_field_list(kwargs, 'exclude_fields')
+        for arg_name in ['fields', 'exclude_fields']:
+            field_args[arg_name] = _normalized_fields_set(kwargs.pop(arg_name, None))
 
         # Instantiate the superclass normally
         super(DynamicFieldsSerializerMixin, self).__init__(*args, **kwargs)
 
         # Accept PARAMS passing from the 'request' in the serializer's context.
-        if hasattr(self, 'context') and isinstance(self.context, dict):
-            request = self.context.get('request', None)
-            # The 'fields' and 'exclude_fields" in request should only apply to
-            # top level serializer.
-            top_level = self.context.get('top_level', True)
-            if request and top_level:
-                fields += request.query_params.getlist('fields', [])
-                exclude_fields += request.query_params.getlist('exclude_fields', [])
```

Comment:
For normal get request with `?fields=a,b` this returns `["a,b"]`.

I remember testing the GET requests previously but apparently I did not!